lib/sycamore/sycamore/transforms/extract_graph.py

```diff
@@ -139,15 +95,58 @@ def finalize(nodes):
                 if "nodes" in doc["properties"]:
                     del doc["properties"]["nodes"]
 
+        doc = Document()
         for value in result["nodes"].values():
-            docs[0]["elements"].append(value)
+            doc["elements"].append(value)
+
+        docs.append(doc)
 
         reader = DocSetReader(docset.context)
         return reader.document(docs)
 
 
-class SupervisedExtractor(GraphExtractor):
-    pass
+class MetadataExtractor(GraphExtractor):
+    """
+    Extracts metadata from documents and represents them as nodes and relationship in neo4j
+
+    Args:
+        metadata: a list of GraphMetadata that is used to determine what metadata is extracted
+    """
+
+    def __init__(self, metadata: list[GraphMetadata]):
+        self.metadata = metadata
+
+    def extract(self, docset: "DocSet") -> "DocSet":
+        docset.plan = ExtractMetadata(docset.plan, self)
+        docset = self.resolve(docset)
+
+        return docset
+
+    def extract_metadata(self, doc: Document) -> Document:
+        """
+        Extracts metadata from documents and stores them in the 'nodes' key of 'properties in each document
+        """
+        nodes: Dict[str, Dict[str, Any]] = {}
+        for m in self.metadata:
+            if m.nodeKey not in doc["properties"]:
```

Comment:
fun note from @alexaryn : you can replace this with

```
key = ...
value = doc["properties"].get(key)
if value is None:
  continue
# remaining logic

```
This results in 1 map read instead of 2. Not really an issue here but interesting